frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateViewModel.kt

```diff
@@ -27,22 +32,68 @@ internal class ExpiryDateViewModel @Inject constructor(
     private val style: ExpiryDateComponentStyle,
 ) : ViewModel() {
 
-    val componentState = ExpiryDateComponentState(inputComponentStateMapper.map(style.inputStyle))
+    val componentState = provideViewState(style.inputStyle)
     val componentStyle = provideViewStyle(style.inputStyle)
 
+    // Flag to determine that component has been already focused before
+    // Needed to prevent validation on focus switch for initial component state
+    private var wasFocused = false
+
+    fun onFocusChanged(isFocused: Boolean) {
+        if (isFocused) wasFocused = isFocused
```

Comment:
Do you need this condition ? . What I can understand is when isFocused is `true` wasFocused will be assigned as `true` and if not default it will be false. Instead you can just assign wasFocused = isFocused ?

Reply:
We need this condition because when we launch this text field, it is by default able to focus which is false. So, need to determine it.
![ss](https://user-images.githubusercontent.com/102042908/187713719-ef65a1c2-f741-4047-b97b-309581b8db39.png)